packages/viz/src/BaseChart/BaseChart.tsx

```diff
@@ -60,7 +60,13 @@ export const HvBaseChart = forwardRef<ReactECharts, HvBaseChartProps>(
       instance.setOption(option, {
         replaceMerge: ["xAxis", "yAxis", "series", "dataset"],
       });
-    }, [theme, option]);
+
+      return () => {
+        // Dispose the instance of the chart when the component unmounts. This ensures that
+        // the chart theme will update when the theme changes.
+        instance.dispose();
+      };
+    }, [theme, option, activeTheme]);
```

Comment:
Do we want `activeTheme` to affect the rest of the behaviour? Couldn't this be a separate `useEffect`?